Add Feedback to Issue Categorizer

grodrigues3 opened the pull request:
Detect when a label added by the bot was later changed by a human being.  Use that information to update the model.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1556)

<!-- Reviewable:end -->

grodrigues3 committed:
Update the Model When Human Fixes Bot Mistake

apelisse commented on mungegithub/mungers/issue-categorizer.go:
```diff
@@ -103,3 +97,59 @@ func (s *LabelMunger) Munge(obj *github.MungeObject) {
+	botEvents := event.FilterEvents(myEvents, event.And([]event.Matcher{event.BotActor(), event.AddLabel{}, event.LabelPrefix(s)}))
+
+	if botEvents.Empty() {
+		glog.Infof("Found no human corrections for issue %d and %s prefix", obj.Issue.Number, s)
```
This log is not correct, and should be below (where there is no log)

apelisse commented on mungegithub/mungers/issue-categorizer.go:
```diff
@@ -103,3 +97,59 @@ func (s *LabelMunger) Munge(obj *github.MungeObject) {
+		}),
+	)
+
+	if humanEventsAfter.Empty() {
```
You should log here

foxish commented:
Review status: 0 of 3 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[mungegithub/mungers/issue-categorizer.go, line 26 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYeIufBhYhCmatc2tT:-KPYeIufBhYhCmatc2tU:b-ub0gw5) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L26)):*
> ```Go
> 	"k8s.io/contrib/mungegithub/github"
> 	"k8s.io/contrib/mungegithub/mungers/matchers/event"
> 	"io/ioutil"
> ```

This import should move up to the top block.

---

*[mungegithub/mungers/issue-categorizer.go, line 30 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhmtshcia4YhfrOOQ:-KPYhmtshcia4YhfrOOR:bjo5uwm) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L30)):*
> ```Go
> 	"github.com/golang/glog"
> 	"github.com/spf13/cobra"
> 	"k8s.io/contrib/mungegithub/features"
> ```

This import should move to the previous block.

---

*[mungegithub/mungers/issue-categorizer.go, line 92 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhASpkHwxObBBjk08:-KPYhASpkHwxObBBjk09:b-u4ct2u) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L92)):*
> ```Go
> 	}
> 	defer routingLabelsToApply.Body.Close()
> 	response, err := ioutil.ReadAll(routingLabelsToApply.Body)
> ```

This should be a separate function for convenience, or one could use that interface for the Model that I mentioned.

---

*[mungegithub/mungers/issue-categorizer.go, line 102 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYi-lpdVhOLDcoiQsK:-KPYi-lpdVhOLDcoiQsL:b-gt3tu5) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L102)):*
> ```Go
> 
> func getHumanCorrectedLabel(obj *github.MungeObject, s string) *string {
> 	myEvents, _ := obj.GetEvents()
> ```

We should be checking for errors and returning in case something in this function fails.

---

*[mungegithub/mungers/issue-categorizer.go, line 143 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYf-72UHJGONB3YvLt:-KPYf-72UHJGONB3YvLu:b80m5id) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L143)):*
> ```Go
> 	if newComponentLabel != nil {
> 		newLabels = append(newLabels, *newComponentLabel)
> 	}
> ```

I would rather split this up into two functions, one to check if an update is needed, and another to apply the update, so that it is less confusing. 
Maybe we could also make the model its own object with an interface:

```
type Model interface {
     NeedsUpdate() ...
     Update() ...
     GetLabel() ...
}
```

Then we could have the config options associated with the model in one place and an easy way to test also.

---

*[mungegithub/mungers/matchers/event/event.go, line 87 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhVXiyZw6bzdRC9zf:-KPYhVXjrE-mGAcDkxO-:bfsjmde) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/matchers/event/event.go#L87)):*
> ```Go
> 
> func MungeBotActor() Matcher {
> 	return Actor("k8s-merge-robot")
> ```

@apelisse Matchers and events are making things quite convenient :)

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

foxish commented:
*[mungegithub/mungers/issue-categorizer.go, line 148 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYjIm4yo8ofY6s6pie:-KPYjIm4yo8ofY6s6pif:bc3fabv) ([raw file](https://github.com/kubernetes/contrib/blob/1761784ad7f58d098e1cb950958480b584fcdf08/mungegithub/mungers/issue-categorizer.go#L148)):*
> ```Go
> 	if len(newLabels) != 0 {
> 		glog.Infof("Updating the models on the server")
> 		_, err := http.PostForm("http://issue-triager-service:5000",
> ```

If we want to port this munger to other repositories, we should have a different service-name, that is per-repo. The URL is being used in two places now and could be moved to a constant. Even better would be to accept it as a flag, so that we can run it on other repos.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

apelisse commented:
*[mungegithub/mungers/matchers/event/event.go, line 87 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhVXiyZw6bzdRC9zf:-KPZTEKMptkXjj22TtdD:b-c1ixi4) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/matchers/event/event.go#L87)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> @apelisse Matchers and events are making things quite convenient :)
</details>
I'm glad you like it :-) more to come!

Very good comments overall

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

grodrigues3 commented:
Review status: 0 of 3 files reviewed at latest revision, 9 unresolved discussions.

---

*[mungegithub/mungers/issue-categorizer.go, line 92 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhASpkHwxObBBjk08:-KPZXbo0Mku6fmfUC8UY:bu2p5l6) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L92)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> This should be a separate function for convenience, or one could use that interface for the Model that I mentioned.
</details>
reading the labels could be a separate function?  It's only used once though.   What is the additional convenience? 

---

*[mungegithub/mungers/issue-categorizer.go, line 102 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYi-lpdVhOLDcoiQsK:-KPZXW_3xOhUVyumawwH:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L102)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> We should be checking for errors and returning in case something in this function fails.
</details>
Done.

---

*[mungegithub/mungers/issue-categorizer.go, line 107 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPTtXcKv5CZ_cfZugBl-r1-107:-KPZXJUIQhNvPQDzw9jb:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L107)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> This log is not correct, and should be below (where there is no log)
</details>
Done.

---

*[mungegithub/mungers/issue-categorizer.go, line 121 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPTtXcKv5CZ_cfZugBl-r1-121:-KPZXIW7vgEDftNqWfAW:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L121)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> You should log here
</details>
Done.

---

*[mungegithub/mungers/issue-categorizer.go, line 148 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYjIm4yo8ofY6s6pie:-KPZWZhK9yoTHeiQ0GG4:bae0prz) ([raw file](https://github.com/kubernetes/contrib/blob/1761784ad7f58d098e1cb950958480b584fcdf08/mungegithub/mungers/issue-categorizer.go#L148)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> If we want to port this munger to other repositories, we should have a different service-name, that is per-repo. The URL is being used in two places now and could be moved to a constant. Even better would be to accept it as a flag, so that we can run it on other repos.
</details>
I could see moving it to a constant, but I'm not sure if there are ever intentions of using an autolabeler on other repos.  Making it more general might be a YAGNI code smell

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

foxish commented:
Review status: 0 of 3 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[mungegithub/mungers/issue-categorizer.go, line 92 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhASpkHwxObBBjk08:-KPZZEDufina7gr2a_7W:b-dyba6t) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L92)):*
<details><summary><i>Previously, grodrigues3 wrote…</i></summary>
> reading the labels could be a separate function?  It's only used once though.   What is the additional convenience? 
</details>
It appears to me that the munge function does too many things, i.e. update the model and apply new labels. Separating them out, or even having the separate object dealing with the Model would make it testable and we could possibly let us change the underlying implementation more freely. 

@apelisse what do you think? Is it overkill?

---

*[mungegithub/mungers/issue-categorizer.go, line 148 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYjIm4yo8ofY6s6pie:-KPZZO69hO6nK7-bvUc6:b-685ayt) ([raw file](https://github.com/kubernetes/contrib/blob/1761784ad7f58d098e1cb950958480b584fcdf08/mungegithub/mungers/issue-categorizer.go#L148)):*
<details><summary><i>Previously, grodrigues3 wrote…</i></summary>
> I could see moving it to a constant, but I'm not sure if there are ever intentions of using an autolabeler on other repos.  Making it more general might be a YAGNI code smell
</details>
I think the eventual plan may be https://github.com/kubernetes/kubernetes/issues/24343, in which case it would make sense to make it capable of running on multiple repos. @bgrant0607 will likely know if there is value in making this more generic.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

apelisse commented:
*[mungegithub/mungers/issue-categorizer.go, line 148 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYjIm4yo8ofY6s6pie:-KPZ_QLgTuXyr6mXD14Q:b-d58nnu) ([raw file](https://github.com/kubernetes/contrib/blob/1761784ad7f58d098e1cb950958480b584fcdf08/mungegithub/mungers/issue-categorizer.go#L148)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> I think the eventual plan may be https://github.com/kubernetes/kubernetes/issues/24343, in which case it would make sense to make it capable of running on multiple repos. @bgrant0607 will likely know if there is value in making this more generic.
</details>
I would certainly make that a flag ..

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

grodrigues3 commented:
Review status: 0 of 3 files reviewed at latest revision, 9 unresolved discussions.

---

*[mungegithub/mungers/issue-categorizer.go, line 26 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYeIufBhYhCmatc2tT:-KPZZLnbn_9uqn8l9Rr_:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L26)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> This import should move up to the top block.
</details>
Done.

---

*[mungegithub/mungers/issue-categorizer.go, line 30 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhmtshcia4YhfrOOQ:-KPZZM3st1wIRqmBDTOw:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L30)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> This import should move to the previous block.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

apelisse commented:
Review status: 0 of 3 files reviewed at latest revision, 9 unresolved discussions.

---

*[mungegithub/mungers/issue-categorizer.go, line 92 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhASpkHwxObBBjk08:-KPZ_arieywK2prcW5Ib:b5xvwso) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L92)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> It appears to me that the munge function does too many things, i.e. update the model and apply new labels. Separating them out, or even having the separate object dealing with the Model would make it testable and we could possibly let us change the underlying implementation more freely. 
> 
> @apelisse what do you think? Is it overkill?
</details>
It could be better organized indeed. It's not awesomely easy to test or understand at first sight how it's organized. That could be left as an exercise though.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

apelisse commented:
*[mungegithub/mungers/issue-categorizer.go, line 53 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPdHwi130JZNlujpO_E:-KPdHwi130JZNlujpO_F:b-cc466m) ([raw file](https://github.com/kubernetes/contrib/blob/501528c049292101b5d836f939fa190a02cb979b/mungegithub/mungers/issue-categorizer.go#L53)):*
> ```Go
> // AddFlags will add any request flags to the cobra `cmd`
> func (lm *LabelMunger) AddFlags(cmd *cobra.Command, config *github.Config) {
> 	cmd.Flags().StringVar(&lm.TriagerUrl, "url", "", "Url on which ml web service is listening")
> ```

The parameter will be "url" which is way too generic. Can we make it "triager-url" or something similar instead?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

apelisse commented:
Review status: 0 of 5 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[mungegithub/mungers/issue-categorizer.go, line 163 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPdINprtZuF5tsO3WYn:-KPdINprtZuF5tsO3WYo:b-w33tgt) ([raw file](https://github.com/kubernetes/contrib/blob/501528c049292101b5d836f939fa190a02cb979b/mungegithub/mungers/issue-categorizer.go#L163)):*
> ```Go
> 		}
> 	}
> }
> ```

Misses extra-line at the end :-)

---

*[mungegithub/submit-queue/deployment.yaml, line 48 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPdIUh-fDt5LBnY_1kf:-KPdIUh-fDt5LBnY_1kg:b753s5n) ([raw file](https://github.com/kubernetes/contrib/blob/501528c049292101b5d836f939fa190a02cb979b/mungegithub/submit-queue/deployment.yaml#L48)):*
> ```YAML
>         - --enable-md-yaml=$(ENABLE_MD_YAML)
>         - --label-file=$(LABEL_FILE)
>         - --triager-url=$(TRIAGER_URL)
> ```

it is triager-url here, I'm puzzled..

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

apelisse commented:
Reviewed 2 of 3 files at r1, 2 of 3 files at r4.
Review status: 4 of 5 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

grodrigues3 commented:
Review status: 2 of 6 files reviewed at latest revision, 12 unresolved discussions.

---

*[mungegithub/mungers/issue-categorizer.go, line 92 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYhASpkHwxObBBjk08:-KPsaY1ibvYlp-t7Y-UW:b-lhgbuk) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L92)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> It could be better organized indeed. It's not awesomely easy to test or understand at first sight how it's organized. That could be left as an exercise though.
</details>
It's reorganized.

---

*[mungegithub/mungers/issue-categorizer.go, line 143 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYf-72UHJGONB3YvLt:-KPsacPSnT18IglITlvV:bb1mowg) ([raw file](https://github.com/kubernetes/contrib/blob/5e470fdbcafa0b2574acb831e1da408f66187b59/mungegithub/mungers/issue-categorizer.go#L143)):*
<details><summary><i>Previously, foxish (Anirudh Ramanathan) wrote…</i></summary>
> I would rather split this up into two functions, one to check if an update is needed, and another to apply the update, so that it is less confusing. 
> Maybe we could also make the model its own object with an interface:
> 
> ```
> type Model interface {
>      NeedsUpdate() ...
>      Update() ...
>      GetLabel() ...
> }
> ```
> 
> Then we could have the config options associated with the model in one place and an easy way to test also.
</details>
Without explicitly creating a Model interface, I implemented these functions so it's more easily testable

---

*[mungegithub/mungers/issue-categorizer.go, line 148 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPYjIm4yo8ofY6s6pie:-KPsat5CKg1SbddTOWel:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/1761784ad7f58d098e1cb950958480b584fcdf08/mungegithub/mungers/issue-categorizer.go#L148)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> I would certainly make that a flag ..
</details>
Done.

---

*[mungegithub/mungers/issue-categorizer.go, line 53 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPdHwi130JZNlujpO_E:-KPsaUTMGfiqj2P_qWjS:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/501528c049292101b5d836f939fa190a02cb979b/mungegithub/mungers/issue-categorizer.go#L53)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> The parameter will be "url" which is way too generic. Can we make it "triager-url" or something similar instead?
</details>
Done.

---

*[mungegithub/mungers/issue-categorizer.go, line 163 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPdINprtZuF5tsO3WYn:-KPsbbdZufpPuiqGwEjm:b-896fix) ([raw file](https://github.com/kubernetes/contrib/blob/501528c049292101b5d836f939fa190a02cb979b/mungegithub/mungers/issue-categorizer.go#L163)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> Misses extra-line at the end :-)
</details>
Done.

---

*[mungegithub/submit-queue/deployment.yaml, line 48 \[r4\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPdIUh-fDt5LBnY_1kf:-KPsbsn1oTITOI15KGPv:bjhnvep) ([raw file](https://github.com/kubernetes/contrib/blob/501528c049292101b5d836f939fa190a02cb979b/mungegithub/submit-queue/deployment.yaml#L48)):*
<details><summary><i>Previously, apelisse (Antoine Pelisse) wrote…</i></summary>
> it is triager-url here, I'm puzzled..
</details>
changed it to triager-url 

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

foxish commented:
Review status: 2 of 6 files reviewed at latest revision, 10 unresolved discussions.

---

*[mungegithub/mungers/issue-categorizer.go, line 84 \[r5\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556#-KPscxnAzAw6EPC1bZbj:-KPscxnAzAw6EPC1bZbk:bg6wen9) ([raw file](https://github.com/kubernetes/contrib/blob/f781038af1536fbbc5c2cb4f7065d7e8298622ac/mungegithub/mungers/issue-categorizer.go#L84)):*
> ```Go
> 			updateModel(lm.TriagerUrl, obj.Issue.Title, obj.Issue.Body, newLabels)
> 		} else{
> 			glog.Info(" no update; no human has corrected our bot!" );
> ```

nit: Maybe we should log this at a higher log level (V6?), or I fear we'll see a lot of these and things will get buried in the noise.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1556)*
<!-- Sent from Reviewable.io -->

foxish commented:
LGTM except for that one nit.

apelisse commented:
/lgtm

apelisse commented:
/lgtm

apelisse commented:
/lgtm
